Client/Frontend/Menu/MenuView.swift

```diff
@@ -335,67 +348,38 @@ class MenuView: UIView {
     }
 }
 
-extension  MenuView: MenuPageViewDelegate {
-    func menuPageView(menuPageView: MenuPageCollectionViewCell, didSelectMenuItem menuItem: MenuItemView, atIndexPath indexPath: NSIndexPath) {
-        menuItemDelegate?.menuView(self, didSelectItemAtIndexPath: indexPath)
-    }
-}
-
 extension MenuView: UICollectionViewDataSource {
     func numberOfSectionsInCollectionView(collectionView: UICollectionView) -> Int {
-        return 1
+        return menuItemDataSource?.numberOfPagesInMenuView(self) ?? 0
     }
 
     func collectionView(collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
-        let numberOfPagesInView = menuItemDataSource?.numberOfPagesInMenuView(self) ?? 0
-        return numberOfPagesInView
+        return menuItemDataSource?.menuView(self, numberOfItemsForPage: section) ?? 0
     }
 
     func collectionView(collectionView: UICollectionView, cellForItemAtIndexPath indexPath: NSIndexPath) -> UICollectionViewCell {
-        let cell = collectionView.dequeueReusableCellWithReuseIdentifier(pagingCellReuseIdentifier, forIndexPath: indexPath) as! MenuPageCollectionViewCell
-        cell.pageIndex = indexPath.row
-        cell.items = itemsForPageIndex(indexPath.row)
-        cell.menuItemDelegate = self
-        cell.numberOfItemsInRow = CGFloat(menuItemDataSource?.numberOfItemsPerRowInMenuView(self) ?? 0)
-        return cell
-    }
-
-    private func itemsForPageIndex(pageIndex: Int) -> [MenuItemView] {
-        var itemsForPageIndex = [MenuItemView]()
-        let numberOfItemsForPage = menuItemDataSource?.menuView(self, numberOfItemsForPage: pageIndex) ?? 0
-        for index in 0..<numberOfItemsForPage {
-            let indexPath = NSIndexPath(forItem: index, inSection: pageIndex)
-            if let view = cachedItems[indexPath] ?? menuItemDataSource?.menuView(self, viewForItemAtIndexPath: indexPath) {
-                if cachedItems[indexPath] == nil {
-                    cachedItems[indexPath] = view
-                }
-
-                itemsForPageIndex.append(view)
+        let cell = collectionView.dequeueReusableCellWithReuseIdentifier(pagingCellReuseIdentifier, forIndexPath: indexPath) as! MenuItemCollectionViewCell
+        if let cellView = menuItemDataSource?.menuView(self, viewForItemAtIndexPath: indexPath) {
```

Comment:
Can we merge the cellView with the MenuItemCollectionViewCell now that we have the single collection view or do we use the view elsewhere in a different context?